module/platform/NUsense/HardwareIO/src/NUsense/SIProcessor.cpp

```diff
@@ -0,0 +1,59 @@
+#include "SIProcessor.hpp"
+
+#include <vector>
+
+namespace NUsense {
+    std::string msg_to_nbs(message::platform::RawSensorsV2& msg) {
+
+        // Use a vector to preprocess the packet then once it is filled,
+        // We insert everything to a string stream
+        std::vector<uint8_t> packets;
+
+        // Add header
+        packet_data.push_back(0xE2);
+        packet_data.push_back(0x98);
+        packet_data.push_back(0xA2);
+
+        // Serialise protobuf message to string then add everything to our vector
+        std::string protobuf_bytes = msg.SerializeToString();
```

Comment:
This is going to be a Neutron not a protobuf, so you need to use the NUClear serialisation method

Reply:
Does the nuclear one agree with the protobuf one? I just wanna make sure as nanopb agrees with vanilla protobuf only